src/arrayMethodJoin.js

```diff
@@ -4,8 +4,35 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let resultString = '';
+
+    if (this.length === 0) {
+      return resultString;
+    }
+
+    if (this.length === 1) {
+      resultString += this[0];
+
+      return resultString;
+    }
```

Comment:
the check for array length being 0 or 1 is not necessary. The loop in the code will handle this case correctly.
```suggestion
```